app/src/main/java/petermcneil/musiclibrary/services/database/ArtistService.java

```diff
@@ -0,0 +1,161 @@
+package petermcneil.musiclibrary.services.database;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.dao.DataAccessException;
+import org.springframework.jdbc.core.ResultSetExtractor;
+import org.springframework.jdbc.core.namedparam.EmptySqlParameterSource;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Service;
+import petermcneil.domain.Artist;
+import petermcneil.domain.Bio;
+import petermcneil.musiclibrary.services.CRUDService;
+
+import java.sql.SQLException;
+import java.sql.ResultSet;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+
+@Service
+public class ArtistService implements CRUDService<Artist> {
+
+    private final NamedParameterJdbcTemplate jdbcTemplate;
+    private final CRUDService<Bio> bioService;
+    private static final Logger LOG = LoggerFactory.getLogger(ArtistService.class);
+
+    public ArtistService(final NamedParameterJdbcTemplate jdbcTemplate, CRUDService<Bio> bio){
+        this.jdbcTemplate = jdbcTemplate;
+        this.bioService = bio;
+    }
+
+
+    @Override
+    public Artist get(Integer artistId) {
+        MapSqlParameterSource params = new MapSqlParameterSource();
+        params.addValue("artistId", artistId);
+        LOG.info("RESPONSE: Returning the artist at the id: {}", artistId);
+        return jdbcTemplate.query("SELECT * FROM artist WHERE idartist=:artistId", params, new ArtistExtractor());
+    }
+
+    @Override
+    public List<Artist> getList() {
+        return jdbcTemplate.query("SELECT * FROM artist", new EmptySqlParameterSource(), new ResultSetExtractor<List<Artist>>() {
+            @Override
+            public List<Artist> extractData(ResultSet rs) throws SQLException, DataAccessException {
+                List<Artist> result = new ArrayList<>();
+                while(rs.next()){
+                    Artist.Builder artist =  Artist.artistBuilder();
+
+                    artist.artistId(rs.getInt("idartist"));
+                    artist.name(rs.getString("name"));
+                    artist.type(getType(rs.getInt("idartisttype")));
+                    artist.bio(bioService.get(rs.getInt("idbio")));
```

Comment:
Just a note for the future when we get everything working this can be made more efficient using db JOINs (remind me this because I'll probably forget)...but keep using this approach for now